wagtail/contrib/modeladmin/options.py

```diff
@@ -370,6 +371,29 @@ def delete_view(self, request, instance_pk):
         view_class = self.delete_view_class
         return view_class.as_view(**kwargs)(request)
 
+    def get_edit_handler(self, instance, request):
```

Comment:
@mr-stateradio I know the original method did all of this in one method, but I feel we'd be missing a trick here if we didn't break things down a little bit further. For example, we could rename this method to `get_bound_edit_handler()` (which would return the same value it does currently), but have it hand over to a `get_edit_handler` method to get/create an `edit_handler` instance, which could in turn hand over to a `get_panels()` method to get/create a `panels` list.

There's certainly a line to be drawn with devolution, but in this case, I feel it would result in much clearer code, greatly improve the opportunities for overriding/extending, and would in turn make tests clearer and easier to write. What are your thoughts?

Reply:
I suppose we could just move the `edit_handler.bind_to_model(self.model)` line back to `ModelFormView.get_edit_handler()`, rather than introduce separate `get_bound_edit_handler()` and `get_edit_handler()` methods.

Reply:
@ababic fixed in https://github.com/wagtail/wagtail/pull/4900/commits/d3baf096e35d840a5c277a7c1fb2a473aa9804f2

I think the structure and responsibilities of the two `get_edit_handler` is much clearer now!